playbooks/ansible-test-changelog/library/validate_changelog.py

```diff
@@ -0,0 +1,211 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# (c) 2022, Aubin Bikouo <@abikouo>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+
+DOCUMENTATION = r"""
+
+module: validate_changelog
+
+short_description: Validate that a pull request contains changelog fragments when required.
+
+author:
+    - "Aubin Bikouo (@abikouo)"
+
+description:
+  - This module validates that each PR defines a valid changelog fragment as requested.
+  - Refer to U(https://docs.ansible.com/ansible/latest/community/development_process.html#creating-changelog-fragments) for
+    more information about changelog.
+
+options:
+  repository:
+    description:
+    - Path to git repository.
+    type: str
+    required: true
+    aliases:
+    - path
+  branch:
+    description:
+    - the default branch to test against.
+    type: str
+    default: main
+"""
+
+EXAMPLES = r"""
+- name: Test changelog against master branch
+  validate_changelog:
+    repository: path_to_the_repository
+    branch: master
+"""
+
+RETURN = r"""
+"""
+
+import re
+import traceback
+import os
+from collections import defaultdict
+
+from ansible.module_utils.basic import AnsibleModule, missing_required_lib
+
+try:
+    import yaml
```

Comment:
I would not bother doing this kind of trick for changes in this repository. In addition `yaml` is supposed to be an Ansible depencency.

Reply:
ok good I will remove it with the rebase of the PR